_data-prepper/pipelines/configuration/sources/documentdb.md

```diff
@@ -0,0 +1,98 @@
+        - collection: my-collection
+          export: true
+          stream: true
+      acknowledgments: true
+```
+{% include copy-curl.html %}
+
+## Configuration
+
+You can use the following options to configure the `documentdb` source.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`host` | Yes | String  | The hostname of the DocumentDB cluster.
+`port` | No | Integer | The port number of the DocumentDB cluster. Defaults to `27017`.
+`trust_store_file_path` | No | String | The path to a truststore file that contains the public certificate for the DocumentDB cluster.
+`trust_store_password` | No | String | The password for the truststore specified by `trust_store_file_path`.
+`authentication` | Yes | Authentication | The authentication configuration. See the [authentication](#authentication) section for more information.
+`collections` | Yes | List | A list of collection configurations. Exactly one collection is required. See the [collections](#collection) section for more information.
+`s3_bucket` | Yes | String  | The Amazon S3 bucket to use for processing events from DocumentDB.
+`s3_prefix` | No | String  | An optional key prefix in Amazon S3. By default, there is no key prefix.
+`s3_region` | No | String  | The AWS region where the Amazon S3 bucket resides.
+`aws` | Yes | AWS | The AWS configuration. See the [aws](#aws) section for more information.
+`id_key` | No | String  | When specified, the `_id` field from DocumentDB is set to the key name specified by `id_key`. You can use this when you need more information beyond the `ObjectId` string saved to your sink. By default, the `_id` is not made as part of the event.
+`direct_connection` | No | Boolean  | When `true`, the MongoDB driver connects directly to the specified DocumentDB server(s) without discovering and connecting to the entire replica set. Defaults to `true`.
+`read_preference` | No | String  | Determines how to read from DocumentDB. See [Read Preference Modes](https://www.mongodb.com/docs/v3.6/reference/read-preference/#read-preference-modes) for details. Defaults to `primaryPreferred`.
+`disable_s3_read_for_leader` | No | Boolean  | When `true`, the current leader node does not read from Amazon S3. It only reads the stream. Defaults to `false`.
+`partition_acknowledgment_timeout` | No | Duration  | Configures the duration for which the node holds a partition. Defaults to `2h`.
+`acknowledgments` | No | Boolean  | When set to `true`, enables [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines#end-to-end-acknowledgments) on this source after events are sent to the sinks.
+`insecure` | No | Boolean | Disables TLS. Defaults to `false`. Do not use this value in production.
+`ssl_insecure_disable_verification` | No | Boolean | Disables TLS hostname verification. Defaults to `false`. Do not enable this flag in production. Instead, use the `trust_store_file_path` to verify the hostname.
+
+### authentication
```

Comment:
I'm assuming that this and the following headings are intentionally lowercase.

Reply:
Yes, this is the configuration settings proper name. I'll set in code format for clarity.